packages/core/src/engine/Autodiff.ts

```diff
@@ -920,14 +1009,31 @@ const compileNode = (
   }
 };
 
+const polyRoots = (coeffs: number[]): number[] => {
+  const n = coeffs.length;
+  // https://en.wikipedia.org/wiki/Companion_matrix
+  const companion = Matrix.zeros(n, n);
+  for (let i = 0; i + 1 < n; i++) {
+    companion.set(i + 1, i, 1);
+    companion.set(i, n - 1, -coeffs[i]);
+  }
+  companion.set(n - 1, n - 1, -coeffs[n - 1]);
+
+  const decomp = new EigenvalueDecomposition(companion);
```

Comment:
So basically the trick here is to construct a square matrix ("companion matrix") for a monic polynomial, whose eigenvalues are roots of this polynomial... right?

A comment would be super helpful for folks who are a little rusty on linear alg (like me :P).